web/src/channel_folders.ts

```diff
@@ -26,7 +26,7 @@ export function initialize(params: StateData["channel_folders"]): void {
 }
 
 export function get_channel_folders(include_archived = false): ChannelFolder[] {
-    const channel_folders = [...channel_folder_by_id_dict.values()].sort((a, b) => a.id - b.id);
+    const channel_folders = [...channel_folder_by_id_dict.values()];
```

Comment:
I feel like this might want to sort by `order`?

Reply:
The API returns the folders sorted by order. And then the map should maintain the insertion order. So, I guess that should be fine.

But yeah might still be better to sort by order here for correctly updating when the order is changed. Will have a look.

Reply:
By typeahead do you mean the dropdown shown in stream settings UI for changing the folder for a channel? It is sorted by name as discussed recently. Can change that as well to sort by order.

Reply:
Yeah, I think it makes sense to do the typeahead sorted by order.